zschema/keys.py

```diff
@@ -30,6 +30,62 @@ def __cmp__(self, other):
             return cmp(int(self.port), int(other.port))
 
 
+def left_merge(lhs, rhs):
+    """
+    Create a dict with all the keys / values of rhs and lhs, where
+    from lhs are used if a key is in both.
+    """
+    ret = rhs.copy()
+    ret.update(lhs)
+    return ret
+
+
+class TypeFactoryFactory(object):
+    """
+    Factory for TypeFactorys (e.g. SubRecordType is a TypeFactory for
+    SubRecords).
+    """
+
+    def __init__(self, cls, args=None, kwargs=None):
+        """
+        This factory acts as a constructor for cls with the first
+        len(args) positional arguments fixed by args, and with kwargs
+        serving as the default keyword arguments.
+        Note: the positional args can be set here, or in the later call,
+        but not in both.
+        """
+        if not callable(cls):
+            raise TypeError("cls must be callable.")
+
+        if args and not isinstance(args, (list, tuple)):
```

Comment:
This should probably be `if args is not None` because otherwise, an empty dict (e.g., `{}`) will not be caught here.

Reply:
I'll get the same effect, and hopefully it will be more readable, by moving the `if args is None: args = ()` up and removing the first condition entirely.